connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala

```diff
@@ -138,12 +140,20 @@ object KustoDataSourceUtils {
       }else if(!applicationCertPath.isEmpty){
         authentication = AadApplicationCertificateAuthentication(applicationId, applicationCertPath, applicationCertPassword);
       }
-    }
-    else if ( {
+    } else if ( {
```

Comment:
This entire section is very messy.
I suggest to rewrite it and split into three sections:
1. Read all the options into local variables
2. Check conflicting options
3. Initialize the authentication handle properly